File: killrvideo-commons/src/main/java/com/killrvideo/utils/IOUtils.java
package com.killrvideo.utils;

import java.net.Socket;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Utilities for network.
 *
 * @author DataStax Developer Advocates Team
 */
public class IOUtils {
    
    /** Logger for Graph. */
    private static final Logger LOGGER = LoggerFactory.getLogger(IOUtils.class);
    
    /**
     * Hide constructor (Utility class).
     */
    private IOUtils() {}
    
    /**
     * Try to open socket on target Adress and port to evaluate if service is running.
     * 
     * @param service
     *      service name (label)
     * @param address
     *      target adress
     * @param port
     *      target port
     * @return
     *      if something is listening on target URL
     */
    public static boolean isServiceReachableAndListening(String service, String address, int port) {
        try (Socket s = new Socket(address, port)) {
            s.setReuseAddress(true);
            LOGGER.info("Connection to {}:{} is working for service {}", address, port, service);
            return true;
        } catch (Exception e) {
            LOGGER.error("Cannot connect to service {} on {}:{} is working for service {}", service, address, port, e);
            return false;
        }
    }

}

File: killrvideo-commons/src/main/kotlin/com/killrvideo/utils/IOUtils.kt
package com.killrvideo.utils

import mu.KotlinLogging
import java.net.Socket

/**
 * Utilities for network.
 *
 * @author DataStax Developer Advocates Team
 */
object IOUtils {
    /** Logger for Graph.  */
    private val logger = KotlinLogging.logger {  }

    /**
     * Try to open socket on target Adress and port to evaluate if service is running.
     *
     * @param service
     * service name (label)
     * @param address
     * target adress
     * @param port
     * target port
     * @return
     * if something is listening on target URL
     */
    fun isServiceReachableAndListening(service: String?, address: String?, port: Int): Boolean {
        try {
            Socket(address, port).use { s ->
                s.reuseAddress = true
                logger.info("Connection to {}:{} is working for service {}", address, port, service)
                return true
            }
        } catch (e: Exception) {
            logger.error("Cannot connect to service {} on {}:{} is working for service {}", service, address, port, e)
            return false
        }
    }
}